hw/bus/i2c/src/i2c.c

```diff
@@ -119,7 +120,10 @@ bus_i2c_read(struct bus_dev *bdev, struct bus_node *bnode, uint8_t *buf,
 
     last_op = !(flags & BUS_F_NOSTOP);
 
-    rc = hal_i2c_master_read(dev->cfg.i2c_num, &i2c_data, timeout, last_op);
+    //rc = hal_i2c_master_read(dev->cfg.i2c_num, &i2c_data, timeout, last_op);
+    rc = i2cn_master_read(dev->cfg.i2c_num, &i2c_data, timeout, last_op,
+                          MYNEWT_VAL(MS5840_I2C_RETRIES));
```

Comment:
I agree, since bus driver is a driver, retries should happen at a higher layer which was i2cn to begin with. I wonder why all the drivers now use bus driver instead of going through i2cn ? Ok, I see since it is a generic bus driver, drivers could directly utilize the bus driver and not worry about which hal to call into SPI or I2C which sounds correct fundamentally but now the bus driver needs to have retries which should be part of the API. Something like 
```
rc = i2cn_master_read(dev->cfg.i2c_num, &i2c_data, timeout, last_op,
                        MYNEWT_VAL(MS5840_I2C_RETRIES));
```

Reply:
i2cn is just a helper which wraps i2c read/write in a loop so kind of a workaround tbh. any retry mechanism in bus driver has to be more configurable and not simply retry on any error since such approach usually just hides actual problems or can just break something.

Reply:
I agree, it does hide the problem but it also helps in recovery at times and make sure the bus gets reset only after a few consecutive attempts instead of resetting right away.

Reply:
what do you mean by "make sure the bus gets reset only after a few consecutive attempts instead of resetting right away"?

Reply:
This was the case before "bus driver" removed the retry logic.